lib/grape/validations/validators/values_validator.rb

```
@@ -85,7 +85,12 @@ def except_message
         end
 
         def required_for_root_scope?
-          @required && @scope.root?
+          return false unless @required
```

fail fast if the `param` is not required.